chore(docs): first iteration of Cawemo's public API (#118)
camunda/camunda-docs-cawemo: content/reference/rest-api/templates/update.md

---
title: 'Templates: update'
weight: 10

menu:
  main:
    name: 'update'
    identifier: 'rest-api-templates-update'
    parent: 'rest-api-templates'
    pre: 'Updates a template.'
---

Updates a template.

# Request

## HTTP Request

```

> PATCH https://cawemo.com/api/v1/templates/:id

Reviewer: We could also offer the publishing as an optional parameter for the create and update actions. This way, users would only need to make one call to update and publish a template.

Author: See https://github.com/camunda/camunda-docs-cawemo/pull/118/files#r687661461

---
title: 'Templates: update'
weight: 10

menu:
  main:
    name: 'update'
    identifier: 'rest-api-templates-update'
    parent: 'rest-api-templates'
    pre: 'Updates a template. This method supports patch semantics.'
---

Updates a template.

> This method supports patch semantics.

Author: With "patch semantics" I mean what you phrased a little differently in https://github.com/camunda/camunda-docs-cawemo/pull/118#discussion_r685952632: Only the properties that you pass in the request body will be updated, the rest stays the same.

Your suggestion in https://github.com/camunda/camunda-docs-cawemo/pull/118#discussion_r685952632 makes this point clear in a simpler way - I'll remove this sentence here and implement your suggestion.

# Request

## HTTP Request

```
PATCH https://cawemo.com/api/v1/templates/:id
```

## Request Body

> In the request body, supply the fields of an [Templates resource]({{< ref "/reference/rest-api/templates/_index.md#resource-representation" >}}) you want to update:

Reviewer 2: ```suggestion
In the request body, supply **only** the properties of a [Templates resource]({{< ref "/reference/rest-api/templates/_index.md#resource-representation" >}}) that you want to update:
```

I would also stick with one term: "properties" or "fields" everywhere.
